File: ecommerce-java/order-service/src/main/java/com/example/orderservice/dto/ResponseOrder.java
package com.example.orderservice.dto;

import com.example.orderservice.domain.OrderEntity;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Getter
@NoArgsConstructor
@AllArgsConstructor
public class ResponseOrder {
    private String productId;
    private Integer qty;
    private Integer unitPrice;
    private Integer totalPrice;

    private String orderId;

    public static ResponseOrder fromResponseEntity(OrderEntity orderEntity) {
        return new ResponseOrder(
                orderEntity.getProductId(),
                orderEntity.getQty(),
                orderEntity.getUnitPrice(),
                orderEntity.getTotalPrice(),
                orderEntity.getOrderId()
        );

    }
}

File: ecommerce-kotlin/user-service/src/main/kotlin/com/example/userservice/dto/ResponseOrder.kt
package com.example.userservice.dto

import java.time.LocalDateTime

data class ResponseOrder(
    val productId: String,
    val qty: Int,
    val unitPrice: Int,
    val totalPrice: Int,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime,
    val orderId: String,
)
